google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/gaxx/retrying/MutateRowsErrorRetryAlgorithm.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.gaxx.retrying;
+
+import com.google.api.core.InternalApi;
+import com.google.api.gax.retrying.BasicResultRetryAlgorithm;
+import com.google.api.gax.retrying.RetryingContext;
+import com.google.api.gax.retrying.TimedAttemptSettings;
+import com.google.cloud.bigtable.data.v2.stub.mutaterows.MutateRowsAttemptResult;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+@InternalApi
+public class MutateRowsErrorRetryAlgorithm
+    extends BasicResultRetryAlgorithm<MutateRowsAttemptResult> {
+  BasicResultRetryAlgorithm<MutateRowsAttemptResult> retryAlgorithm;
+
+  public MutateRowsErrorRetryAlgorithm(
+      BasicResultRetryAlgorithm<MutateRowsAttemptResult> retryAlgorithm) {
+    this.retryAlgorithm = retryAlgorithm;
+  }
+
+  @Override
+  public TimedAttemptSettings createNextAttempt(
+      Throwable previousThrowable,
+      MutateRowsAttemptResult previousResponse,
+      TimedAttemptSettings previousSettings) {
+    return retryAlgorithm.createNextAttempt(previousThrowable, previousResponse, previousSettings);
+  }
+
+  @Override
+  public TimedAttemptSettings createNextAttempt(
+      RetryingContext context,
+      Throwable previousThrowable,
+      MutateRowsAttemptResult previousResponse,
+      TimedAttemptSettings previousSettings) {
+    return retryAlgorithm.createNextAttempt(
+        context, previousThrowable, previousResponse, previousSettings);
+  }
```

Comment:
please move the dumb delegates done as they arent interesting and pull the actual relevant overrides up